### DISCUSSION TITLE
Lambda + Xray and how to propagate the information to S3

### DISCUSSION THREAD

**Discussion by Chandrian** [awslabs/aws-sdk-rust#751] [Q&A] [answered]
Hello, trying again (after almost a year) to set up my lambda function to use Xray. 

1) The lambda function has the Xray enabled
2) In the service map, I can see the initialization and the invocations
3) I can't see any requests to other services like S3 using ListObjectsV2

What I tried: 
<img width="993" alt="image" src="https://user-images.githubusercontent.com/3663566/223537887-f3db1468-3cba-4724-a0a4-c9566454412f.png">

I'm getting the Trace ID from the environment variable and injecting it as value in the header as I understood from [Feature: Customizable Operations ](https://github.com/awslabs/smithy-rs/blob/main/design/src/rfcs/rfc0017_customizable_client_operations.md) by @Velfi .

**Comment by Chandrian** [comment]
Header's name are supposed to be case *in*sensitive. I'm wondering why the HeaderName::from_static is panicking when trying to add a name of "X-Amzn-Trace-Id"? I was just wondering if it would make any difference but because of this I can't try

**Reply by rcoh** [reply]
because `HeaderName::from_static` is a `const fn` you need to provide a lower case header.

**Comment by mpetri** [comment]
From my understanding the issue is described in this blog post: https://wrzasq.pl/blog/x-ray-tracing-for-aws-rust-sdk-in-lambda-runtime.html

the lambda runtime has a demon running to receive traces but there is no rust functionality to send stuff to the demon easily. 

There is an open issue associated with this but in general the behavior is quite sub-standard from my view.

Trying to make this also work I have so far wasted several ours as the rust + xray tracing story is quite bad.

**Reply by handelcamilo** [reply]
There is no way to use Rust lambda with Xray yet?

**Reply by jmklix** [reply, marked as answer]
@handelcamilo not yet, we do have a feature request for that: https://github.com/awslabs/aws-sdk-rust/issues/642
